### ISSUE TITLE
[WC 4.8 beta + PHP 8.0RC3] Atomic: Critical error when trying to select a theme other than current during OBW

### ISSUE THREAD

**Issue by juliaamosova** [woocommerce/woocommerce#28327]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

I tested this issue on an Atomic ephemeral site with WC 4.8 beta + PHP 8.0RC3. Nothing was installed on Atomic (no checkboxes were selected when I created a site). When the user gets to step in Onboarding Wizard to select a theme other than the current one, an error occurs on the site without an option to proceed (error seen in the error logs of Atomic site via ssh):

```
2020-11-17T19:18:01+00:00 CRITICAL Uncaught ArgumentCountError: array_merge() does not accept unknown named parameters in /srv/htdocs/__wp__/wp-includes/widgets.php:1265
Stack trace:
#0 /srv/htdocs/__wp__/wp-includes/widgets.php(1265): array_merge(wp_inactive_widgets: Array, sidebar-1: Array, header-1: Array, footer-1: Array, footer-2: Array, footer-3: Array, footer-4: Array)
#1 /srv/htdocs/__wp__/wp-includes/widgets.php(1220): retrieve_widgets(true)
#2 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(289): _wp_sidebars_changed('Twenty Twenty')
#3 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array)
#4 /srv/htdocs/__wp__/wp-includes/plugin.php(478): WP_Hook->do_action(Array)
#5 /srv/htdocs/__wp__/wp-includes/theme.php(3267): do_action('after_switch_th...', 'Twenty Twenty', Object(WP_Theme))
#6 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(287): check_theme_switched('')
#7 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array)
#8 /srv/htdocs/__wp__/wp-includes/plugin.php(478): WP_Hook->do_action(Array)
#9 /srv/htdocs/__wp__/wp-settings.php(546): do_action('init')
#10 /srv/htdocs/wp-config.php(81): require_once('/srv/htdocs/__w...')
#11 /srv/htdocs/__wp__/wp-load.php(42): require_once('/srv/htdocs/wp-...')
#12 /srv/htdocs/__wp__/wp-cron.php(44): require_once('/srv/htdocs/__w...')
#13 {main}
  thrown in /srv/htdocs/__wp__/wp-includes/widgets.php on line 1265
```

**Expected behavior**

I expected to be able to select a theme other than existing during the onboarding wizard. 

**Steps to reproduce the bug:**

Steps to reproduce the bug:

1. Create an Atomic ephemeral site. Don't select any checkboxes to install extra features. But do select PHP 8.0. 
2. Upload WC 4.8 beta1. 
3. Start Onboarding Wizard. 
4. It doesn't matter what details you'd fill in throughout onboarding.
5. On the theme selection step, select `Live Demo` next to `Storefront` theme for example:

![1](https://user-images.githubusercontent.com/19143190/99441654-2dbecc00-28e6-11eb-8cd5-f6c5eda61a92.jpg)

6. Next, click on the `Choose` button and observe errors that start coming in Console. You won't be able to get passed that stage:

![2](https://user-images.githubusercontent.com/19143190/99441710-462ee680-28e6-11eb-9e7f-4f94325f745f.jpg)

![Screen Shot 2020-11-17 at 2 11 27 PM](https://user-images.githubusercontent.com/19143190/99441736-4fb84e80-28e6-11eb-84cf-1a26d0f06461.png)

![Screen Shot 2020-11-17 at 2 12 10 PM](https://user-images.githubusercontent.com/19143190/99441748-53e46c00-28e6-11eb-8aa1-b287cdf36d43.png)

If you try to just get back to see the site, there will be a message letting you know that there is a Critical Error. You won't be able to view the site:

![Screen Shot 2020-11-17 at 2 14 49 PM](https://user-images.githubusercontent.com/19143190/99441797-6494e200-28e6-11eb-9959-d7f42171df11.png)

**Screenshots**

See above. 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by juliaamosova** [comment]
Noting that this issue could be related to p9o2xV-185-p2 but I did not have `wpcomsh` set up on the site.

**Comment by brandonpayton** [comment]
Hi @juliaamosova,

This is a known issue in core that has been fixed for the WordPress 5.6:
https://core.trac.wordpress.org/changeset/48839

It was fixed under a big PHP 8.0 compatibility ticket:
https://core.trac.wordpress.org/ticket/50913

**Comment by juliaamosova** [comment]
Thank you for your help with this issue @brandonpayton!

**Comment by aidvu** [comment]
You can also download latest WP and extract it to test. Might be worth confirming, instead of waiting for WP 5.6.

Think this should work. :)

```
wget https://github.com/WordPress/WordPress/archive/master.zip
mv __wp__ __wp__old
tar -xzvf master.zip
mv WordPress-master __wp__
```

**Comment by juliaamosova** [comment]
@aidvu thanks! And yep, we are going to test this to make sure everything works with WP 5.6. We've been using the following so far to get WP:

```
wget https://wordpress.org/wordpress-5.6-RC1.zip
unzip wordpress-5.6-RC1.zip
cp -r wordpress/* __wp__
```

But I think your commands will always pool the latest WP dev version without the need to specify the version which is more convenient.

**Comment by aidvu** [comment]
Right. Unzip, not tar. :)

**Comment by tammullen** [comment]
I tried to reproduce using the same setup of Vanilla ephemeral atomic with PHP 8.0 RC3, WooCommerce 4.8 Beta 1 but this time with the new WordPress 5.6 RC1.

I was able to Choose and install Storefront from the Live Demo page with no related console errors (those in my screenshot are related to a known Jetpack issue) or site errors.

![Screenshot_2020-11-18_at_20_11_47](https://user-images.githubusercontent.com/24649833/99583699-d26f0580-29db-11eb-9c49-4c2ea0543b76.png)

Closing as this seems to confirm it is fixed by the latest version of WordPress.
